commons-geometry-euclidean/src/test/java/org/apache/commons/geometry/euclidean/threed/Bounds3DTest.java

```diff
@@ -125,29 +125,12 @@ public void testFrom_invalidBounds() {
         final Vector3D negInf = Vector3D.of(1, 1, Double.NEGATIVE_INFINITY);
 
         // act/assert
-        GeometryTestUtils.assertThrows(() -> {
-            Bounds3D.from(Vector3D.NaN);
-        }, IllegalStateException.class, INVALID_BOUNDS_PATTERN);
-
-        GeometryTestUtils.assertThrows(() -> {
-            Bounds3D.from(Vector3D.POSITIVE_INFINITY);
-        }, IllegalStateException.class, INVALID_BOUNDS_PATTERN);
-
-        GeometryTestUtils.assertThrows(() -> {
-            Bounds3D.from(Vector3D.NEGATIVE_INFINITY);
-        }, IllegalStateException.class, INVALID_BOUNDS_PATTERN);
-
-        GeometryTestUtils.assertThrows(() -> {
-            Bounds3D.from(good, nan);
-        }, IllegalStateException.class, INVALID_BOUNDS_PATTERN);
-
-        GeometryTestUtils.assertThrows(() -> {
-            Bounds3D.from(posInf, good);
-        }, IllegalStateException.class, INVALID_BOUNDS_PATTERN);
-
-        GeometryTestUtils.assertThrows(() -> {
-            Bounds3D.from(good, negInf, good);
-        }, IllegalStateException.class, INVALID_BOUNDS_PATTERN);
+        assertThrows(IllegalStateException.class, () -> Bounds3D.from(Vector3D.NaN));
```

Comment:
We're losing the assertion of the message content here and other places that use the regex pattern assertion. Is there a way to avoid that?

Reply:
HI @darkma773r 
to be honest i don't know. Do we really need those messages? They give us extra info? 
Do they contribute something when executing a test or validating any functionality?

Reply:
`assertThrows` returns the exception instance, which can be used to validate the message pattern.

Reply:
Easier thing might to be to just replace the implementation of `GeometryTestUtils#assertThrows` with JUnit5.

```java
    public static <T extends Throwable> void assertThrows(
        final Executable r, final Class<T> exceptionType, final String message) {
        Optional.ofNullable(Assertions.assertThrows(exceptionType, r).getMessage())
            .ifPresent(excMsg -> Assertions.assertEquals(message, excMsg));
    }
```

Reply:
@darkma773r Also, I think there is confusion about the 3rd parameter. In JUnit5, the 3rd parameter is message for debugging. Here, its the expected message of the expected exception. Perhaps the method name should be clarified to something like `assertThrowsWithMessage`.

Reply:
Hi @darkma773r 
 I still think that messages don't give us much. As @singhbaljit  say it's just a debugging message. 
The first approach was just migrate to JUNit 5. But I agree that if we migrate we should opt for the assertThrows pattern

Reply:
> I still think that messages don't give us much.

The assertions on the expected exception messages are important because they verify not only what was thrown but why and ensure that the exceptions produced by the library are accurate and useful. To remove these assertions after they are already in place would be taking a step backward.

@singhbaljit, is correct in that I was confused on the 3rd parameter to the JUnit 5 `assertThrows` method: I thought it contained the expected message of the thrown exception and not simply a string to display when the assertion failed. As such, `Assertions.assertThrows` is not the complete drop-in replacement for `GeometryTestUtils.assertThrows` that I thought it was. 

Moving forward, the cleanest approach I see here is the following:
1. Remove the two argument `GeometryTestUtils.assertThrows(Runnable, Throwable)` method. All current invocations of this method can be directly replaced with the standard JUnit method.
2. Rename `GeometryTestUtils.assertThrows(Runnable, Throwable, String)` and `GeometryTestUtils.assertThrows(Runnable, Throwable, Pattern)` to `assertThrowsWithMessage` for clarify and implement internally with JUnit assertions, as described by @singhbaljit.

Reply:
I agree with @darkma773r's two suggestions. I would also suggest the following:
1. Remove the `message != null` check for the message assertion in `GeometryTestUtils. assertThrowsWithMessage(Runnable, Throwable, String)`. Instead of using `null` to skip the test (which I assume was done to make `GeometryTestUtils.assertThrows(Runnable, Throwable)` work), it is a validate test to check if the message is in fact `null`.
2. Similarly, I also don't think the pattern in `GeometryTestUtils.assertThrows(Runnable, Throwable, Pattern)` should be optional/nullable.

Reply:
Good points, @singhbaljit. I agree.

Reply:
OK guys let me see what  i can do.

Reply:
HI guys,
I have created a new PR with all the requested changes.

https://github.com/apache/commons-geometry/pull/121